Development

aschiffler opened the pull request:
Weekend - Sprintend 😎
Time for code review

## fixes and development
- eeprom alignment fixed (thx @klahus1)
- mqtt reconnect improvements (thx @tastendruecker123 , @HorstG-57 )
- simple command scheduler (only 1 place fifo); DevControl has still priority because it is no "Info Command"
- InfoAll Command Parser and output to mqtt
- mqtt payload handling bug exploration and fix (thx @klahus1, silverserver)
- documentation and issue management (thx @DanielR92 @stefan123t @lumapu  )
- new workflow to build github release with included "User Manual"
- and things which are not mentioned here (thx @fila612 and others)



## known issues
- missing parser implementation

aschiffler committed:
fix mqtt payload for reactive power limit

aschiffler committed:
Merge branch 'main' into development

aschiffler committed:
small fix

aschiffler committed:
button on index to get FW-Version

aschiffler committed:
write to dtu eeprom  if pwr limit is persistent

aschiffler committed:
Merge branch 'main' into development

DanielR92 committed:
Added a collection of inverter inforamtion

Just to see later the differenc between inverters

DanielR92 committed:
Update Firmware colelction

DanielR92 committed:
Update

DanielR92 committed:
Update for better reading

DanielR92 committed:
little Update

aschiffler committed:
Merge pull request #166 from DanielR92/patch-8

Added a collection of inverter information

DanielR92 committed:
Added new col. and insert new sec. inv eeprom23

stefan123t committed:
Merge pull request #168 from DanielR92/patch-9

Added new col. and insert new sec. inv eeprom23

fila612 committed:
add FW-Data fila612

aschiffler committed:
possible fix ref issue #162

aschiffler committed:
Update User Manual

aschiffler committed:
Merge branch 'development' into dev_private

aschiffler committed:
Merge pull request #170 from fila612/dev_private

add FW-Data fila612

klahus1 committed:
fix possible problems in struct (byte) alignment

lumapu committed:
Merge pull request #175 from klahus1/development_klahus1

fix alignment

aschiffler committed:
complete rework of release workflow add CHANGES.md

aschiffler committed:
Merge branch 'development' of https://github.com/grindylow/ahoy into development

aschiffler committed:
correct path for CHANGES.md

tastendruecker123 committed:
MQTT resub only when reconnected successful

tastendruecker123 committed:
MQTT subscribe only when reconnect successful

aschiffler committed:
Merge pull request #179 from tastendruecker123/mqttfix

MQTT resubscribe only when reconnected successfully, don't call reconnect() when client is still connected (Issue #176)

aschiffler committed:
added comment about mqtt reconnect

aschiffler committed:
removed change power limit on dtu startup

aschiffler committed:
updated Changes.md

lumapu committed:
fixed setup layout for mobile devices

tastendruecker123 committed:
ESP32 support added

* ESP32 adjustments, compiles and runs
* Changed gitignore to ignore debug log files

aschiffler committed:
Merge PR #181 from tastendruecker123/esp32

ESP32 support added

stefan123t commented on tools/esp8266/defines.h:
```diff
@@ -109,15 +109,16 @@ typedef enum { // ToDo: to be verified by field tests
     char topic[MQTT_TOPIC_LEN];
-} mqttConfig_t;
-
+} /*__attribute__((__packed__))*/ mqttConfig_t;
```
Wollten wir das nicht MIT attribute pack aligned(1) weil es leserlicher ist und es ggf auch der compiler sieht nicht nur der präprozessor ?

stefan123t commented on tools/esp8266/hmSystem.h:
```diff
@@ -19,7 +19,8 @@ class HmSystem {
         typedef BUFFER BufferType;
         BufferType BufCtrl;
-        InfoCmdType InfoCmd;
+        InfoCmdType NextInfoCmd; // For cmd-queue FIFO with one place
```
Nextinfocmd ist doch der punkt an dem die API queue in den normalen prozess mündet

stefan123t reviewed (approved):
👍Dann ist der Fix fürs Setup endlich in main!

Habe zwei Punkte kommentiert bzw die könnten wir noch besser beschreiben

aschiffler committed:
added command queue

+ command queue
+ feedback from inverter for actual power limit via InfoCmd -> SystemConfigPara (0x05)
+ REST API will enqueue a new info command
+ change in power limit will enqueue infocm to get actual power limit
+ actual power limit is available under mqtt topic <TOPIC>/<NAME>/ch0/PowerLimit ALWAYS in percent
+ Firmware information will be requested automatically up on start of dtu

aschiffler committed:
Update CHANGES.md
